### DISCUSSION TITLE
Hi I'm getting  Application error  during deploy the main branch

### DISCUSSION THREAD

**Discussion by BilalN10** [saran-surya/email_auth_node#52] [Q&A]
An error occurred in the application and your page could not be served. If you are the application owner, [check your logs for details](https://devcenter.heroku.com/articles/logging#view-logs). You can do this from the Heroku CLI with the command
heroku logs --tail

**Comment by saran-surya** [comment]
Do you have a free tier or a paid account in heroku ?

**Reply by BilalN10** [reply]
Hi.
First of all thanks for reply. I think its free account because I just give my credit card info nothing more than that. And I resolve this error that I mention before but now I'm getting new error. Everything setup well and also got welcome text on web page after deploy it to main branch.
But when I send Otp it gives me  :
I/flutter (15392): email-auth >> Initialising Email-Auth server
I/flutter (15392): email-auth >> Remote server is not available -- using test server --
I/flutter (15392): email-auth >> ❗ Warning this is not reliable on production
I/flutter (15392): --- Package Error ---
I/flutter (15392): Unable to access server. 😑
I/flutter (15392): --- End Package Error ---
I/flutter (15392): Otp send false
I/flutter (15392): --- Package Error ---
I/flutter (15392): Unable to access remote server. 😑
I/flutter (15392): --- End Package Error ---
I/flutter (15392):
I/flutter (15392): --- package Error ---
I/flutter (15392): email-auth >> The remote server is not a valid.
I/flutter (15392): email-auth >> configured server : "https://mail-abot.herokuapp.com"
I/flutter (15392): --- package Error ---

I'm calling this function
void sendOtp() async {
   
    emailAuth = EmailAuth(sessionName: 'Sample session');
    var remeoteServerConfiguration = {
      "server": 'https://mail-abot.herokuapp.com',
      'serverKey': "jAPCMS"
    };
    emailAuth!.config(remeoteServerConfiguration);
    await emailAuth!
        .sendOtp(recipientMail: "my email", otpLength: 6)
        .then((value) {
      print('Otp send $value');
    }).catchError((e) {
      print('error is ${e.toString()}');
    });
  
  }

**Reply by saran-surya** [reply]
Actually visiting the page, seems like the deploy did not happen, you should actually see a home page describing EMAIL_AUTH.
Kindly check if deploy did happen successfully

**Reply by BilalN10** [reply]
can you please tell me my sever url is correct?

**Reply by BilalN10** [reply]
sorry my server url was not correct so change the url now it says

I/flutter (23169): email-auth >> Initialising Email-Auth server
I/flutter (23169): email-auth >> Remote server is not available -- using test server --
I/flutter (23169): email-auth >> ❗ Warning this is not reliable on production
I/flutter (23169): --- Package Error ---
I/flutter (23169): Unable to access server. 😑
I/flutter (23169): --- End Package Error ---
I/flutter (23169): Otp send false
I/flutter (23169): email-auth >> The remote server configurations are valid

my remote Server Configuration are :
 var remoteServerConfiguration = {
      "server": "https://powerful-reaches-50418.herokuapp.com",
      "serverKey": "fnpcoY"
    };

**Reply by saran-surya** [reply]
Are you still facing issues regarding the server ?
If not will be marking it as <b>STALE</b>

**Reply by itsRenad** [reply]
I did not check it again. Actually, I was working on one of my client's projects he cancel the order so I stop my work.

**Reply by saran-surya** [reply]
Okay then, Will mark it as complete then, Thankyou for the info.

**Comment by BilalN10** [comment]
I'm getting this message on web page 
"Hello Welcome to the homepage".

**Reply by saran-surya** [reply]
Can you please login to the terminal of the app, and then perform the request so we can have all the logs noted. Or please put in the command ```npx heroku logs --tail``` and please post them here

**Reply by BilalN10** [reply]
it show me this when I run the command npx heroku logs --tail 
 Error: Missing required flag:
 »    -a, --app APP  app to run command against
 »   See more help with --help

**Reply by saran-surya** [reply]
You should pass in the app name, please read the error logs and documentation 👍

**Reply by saran-surya** [reply]
Also the query in your case will be <br/> ```npx heroku logs --tail -a powerful-reaches-50418``` this should give out the complete logs for the app atleast the live buffer of the latest ones, then you can perform the request and then during the crash we can figure it out. Also Google will not work, u should work with outlook for the case. The changes have been made to the repo so please pull the latest changes.

**Comment by saran-surya** [comment]
Update there is an issue with the external packages from pip. Will be trying to rectify them in the coming weeks. Kindly expect a delay.


Converting this discussion to an issue with added details.
But kindly post the heroku logs will help in identifying the potential future errors.
